src/Analyzer/CodeAnalyzer.cs

```diff
@@ -65,6 +68,12 @@ public CodeAnalyzer() : base(csharpExtension)
             diagnosticsRunner = new DiagnosticsRunner(GetAnalyzers(), GetUtilityAnalyzers(), additionalFiles.ToArray());
         }
 
+        ~CodeAnalyzer()
+        {
+            // delete created temporary directory
+            Directory.Delete(Path.GetDirectoryName(tmpSonarLintPath), true);
```

Comment:
> Isn't it a file instead of a directory?

It's the directory because I'm calling `Path.GetDirectoryName` to get it. I'll explicitly use the directory name.


> Did you explore the possibility to use `using` construct instead of deleting the directory by hand?

In this case, `using` is not the way to do it, because the object should be a `Disposable` and I'm doing that task on the dtor, (`~CodeAnalyzer`).